pkg/pool-manager/pool.go

```diff
@@ -52,10 +53,11 @@ var log = logf.Log.WithName("PoolManager")
 var now = func() time.Time { return time.Now() }
 
 type PoolManager struct {
-	kubeClient       kubernetes.Interface // kubernetes client
-	rangeStart       net.HardwareAddr     // fist mac in range
-	rangeEnd         net.HardwareAddr     // last mac in range
-	currentMac       net.HardwareAddr     // last given mac
+	cachedKubeClient client.Client
+	kubeClient       client.Client
```

Comment:
why are we moving to client.Client?
I'm not against, just want to understand why

Reply:
to keep code that call client the same independently if it's using the cached versin or not.

An alternative is to use the cached client but with Selectors so we specify what we want to cache and what not.